File: platform/packages_py/env_resolver/env_resolver/env_sonarqube_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class SonarqubeEnv:
    api_token: str | None
    base_url: str


def resolve_sonarqube_env(config: dict[str, Any] | None = None) -> SonarqubeEnv:
    return SonarqubeEnv(
        api_token=resolve(None, ['SONAR_API_TOKEN', 'SONAR_TOKEN', 'SONARQUBE_TOKEN'], config, 'api_token', None),
        base_url=resolve(None, ['SONAR_BASE_URL'], config, 'base_url', 'https://sonarcloud.io'),
    )

File: platform/packages_mjs/env-resolver/src/env-sonarqube.x.ts
import { resolve } from '@internal/env-resolve';

export interface SonarqubeEnv {
    apiToken: string | undefined;
    baseUrl: string;
}

export function resolveSonarqubeEnv(config?: Record<string, any>): SonarqubeEnv {
    return {
        apiToken: resolve(undefined, ['SONAR_API_TOKEN', 'SONAR_TOKEN', 'SONARQUBE_TOKEN'], config, 'apiToken', undefined),
        baseUrl: resolve(undefined, ['SONAR_BASE_URL'], config, 'baseUrl', 'https://sonarcloud.io'),
    };
}
